bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandThermostatHandler.java

```diff
@@ -58,28 +58,34 @@ void touchWandUnitHandleCommand(Command command) {
         if (touchWandBridgeHandler != null) {
             if (command instanceof OnOffType) {
                 touchWandBridgeHandler.touchWandClient.cmdThermostatOnOff(unitId, (OnOffType) command);
-            } else {
-                String sCommand = command.toString();
-                switch (sCommand) {
-                    case "cool":
-                    case "heat":
-                    case "fan":
-                    case "auto":
-                    case "dry":
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatMode(unitId, sCommand);
-                        break;
-                    case "low":
-                    case "medium":
-                    case "high":
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatFanLevel(unitId, sCommand);
-                        break;
-                    case "fanAuto":
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatFanLevel(unitId, "auto");
-                        break;
-                    default:
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatTargetTemperature(unitId, sCommand);
-                        break;
-                }
+                return;
+            }
+            if (command instanceof QuantityType) {
```

Comment:
Consider if method should be extended to include the channel also. It seems that currently the channel is assumed from the type of command only. So let's say the someone would try to set **roomTemperature**, wouldn't this silently be accepted as a command for setting **targetTemperature**?

Additionally, for channel **targetTemperature** it would probably also make sense to support DecimalType/StringType using default unit?

Reply:
it do make sense , I will think how I do it and implement across the Things handlers 
Since it is a bit more risky I will do it in a separate PR in the near future 
At the moment I want to make sure the current bug is resolved

Reply:
I have checked temperatureItem.sendCommand("21") and temperatureItem.sendCommand(21) and both are working well with the current implementation